app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/WorkoutDetailsActivity.java

```diff
@@ -1,60 +1,97 @@
+    }
+    public static final String EXTRAS_WORKOUT_ID = qualify("DEVICE_ADDRESS");
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_workout_details);
 
-        //In production version, this ID (or startdate) will be passed from the
-        //past work/workout history s
-        int id = 10;
+        Intent intent = getIntent();
+        mWorkout_id = intent.getIntExtra(EXTRAS_WORKOUT_ID,0);
+
         // In the final version, use R.string.<string id> for titles
         getSupportActionBar().setTitle("Workout Details");
 
         mNavBar.init();
 
         mWorkoutViewModel = ViewModelProviders.of(this).get(WorkoutViewModel.class);
-        Workouts workout = new Workouts();
+        Workout workout = new Workout();
+
+        mDeviceViewModel = ViewModelProviders.of(this).get(DeviceViewModel.class);
+        Device d = new Device("Test_Device","00:11:00:33:AA",-25);
+
+
 
+        mDeviceViewModel.insert(d);
+        //This code maintains a known entry in the DB for testing, the insert will update the
+        //workout with ID 10 in the DB.
         workout.setID(10);
         workout.setPoolLength(25);
         workout.setTotalDistanceTraveled(1200);
-        workout.setStartDateTime(TimestampConverter.fromTimestamp("2018-4-25 12:00:00.000"));
-        workout.setEndDateTime(TimestampConverter.fromTimestamp("2018-4-25 14:00:00.000"));
+        workout.setStartDate(TimestampConverter.fromTimestamp(1554904800));
+        workout.setEndDate(TimestampConverter.fromTimestamp(1554912000));
+        workout.setDeviceMAC(d.getMacAddress());
         workout.setLaps(34);
-        workout.setPoolUnits("Yards");
-        //While we are in development phase, use this to insert a workout that you wish to view.
-        //Uncomment line to add to db
+        workout.setPoolUnits("Meters");
         mWorkoutViewModel.insert(workout);
 
-        workout = mWorkoutViewModel.getWorkoutByID(id);
+        //Below code is commented so Neil has an example of how the new viewmodels and their functions work.
+        //If you comment below lines, it will insert a new workout with the next available unused ID number.
+        //getAllWorkouts and getWorkoutsByDateRange are also below to show how they work, and that they do work.
+        //All this can be deleted at a later date
+
+        /*
+        Workout a_workout = new Workout();
+        a_workout.setPoolLength(50);
+        a_workout.setTotalDistanceTraveled(500);
+        a_workout.setStartDate(TimestampConverter.fromTimestamp(1523440800));
+        a_workout.setEndDate(TimestampConverter.fromTimestamp(1523448000));
+        a_workout.setDeviceMAC(d.getMacAddress());
+        a_workout.setLaps(10);
+        a_workout.setPoolUnits("Meters");
+        mWorkoutViewModel.insert(a_workout);
+
+
+        ArrayList<Workout> test = mWorkoutViewModel.getAllWorkouts();
```

Comment:
Again, it's usually good practice to use `List<>` instead of `ArrayList<>` when declaring a variable.